otx/algorithms/common/adapters/mmcv/utils/config_utils.py

```diff
@@ -180,6 +186,145 @@ def patch_runner(config: Config):
         config.runner.type = "IterBasedRunnerWithCancel"
 
 
+def patch_fp16(config: Config):
+    """Remove FP16 config if running on CPU device and revert to FP32.
+
+    Please refer https://github.com/pytorch/pytorch/issues/23377
+    """
+    if not torch.cuda.is_available() and "fp16" in config:
+        logger.info("Revert FP16 to FP32 on CPU device")
+        if isinstance(config, Config):
+            del config._cfg_dict["fp16"]  # pylint: disable=protected-access
+        elif isinstance(config, ConfigDict):
+            del config["fp16"]
+
+
+def update_basic_hooks(config: Config):
+    """Update basic hooks for OTX trainig.
+
+    This function can be removed by add custom hook cfg into recipe.py directly.
+    """
+    # default adaptive hook for evaluating before and after training
+    add_custom_hook_if_not_exists(
+        config,
+        ConfigDict(
+            type="AdaptiveTrainSchedulingHook",
+            enable_adaptive_interval_hook=False,
+            enable_eval_before_run=True,
+        ),
+    )
+    # Add/remove adaptive interval hook
+    if config.get("use_adaptive_interval", False):
+        update_or_add_custom_hook(
+            config,
+            ConfigDict(
+                {
+                    "type": "AdaptiveTrainSchedulingHook",
+                    "max_interval": 5,
+                    "enable_adaptive_interval_hook": True,
+                    "enable_eval_before_run": True,
+                    **config.pop("adaptive_validation_interval", {}),
+                }
+            ),
+        )
+    else:
+        config.pop("adaptive_validation_interval", None)
+
+    if "early_stop" in config:
+        remove_custom_hook(config, "EarlyStoppingHook")
+        early_stop = config.get("early_stop", False)
+        if early_stop:
```

Comment:
How about to align with above statement?
```suggestion
    if config.get("early_stop", False):
        remove_custom_hook(config, "EarlyStoppingHook")
        if config.early_stop:
```

Reply:
This change requires additional changes at below lines such like `patience=early_stop.patience,` => `patience=config.early_stop.patience,`